services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/SwitchDeleteSpec.groovy

```diff
@@ -1,281 +1,194 @@
-        northbound.deleteSwitch(switchId, false)
+    def "Unable to delete a nonexistent switch"() {
+        when: "Try to delete a nonexistent switch"
+        northbound.deleteSwitch(NON_EXISTENT_SWITCH_ID, false)
 
-        then: "Got 404 NotFound"
+        then: "Get 404 NotFound error"
         def exc = thrown(HttpClientErrorException)
         exc.rawStatusCode == 404
     }
 
-    def "Cannot delete active switch"() {
+    def "Unable to delete an active switch"() {
         given: "An active switch"
         def switchId = topology.getActiveSwitches()[0].dpId
 
-        when: "Try to delete active switch"
+        when: "Try to delete the switch"
         northbound.deleteSwitch(switchId, false)
 
-        then: "Got 400 BadRequest because switch must be deactivated"
+        then: "Get 400 BadRequest error because the switch must be deactivated first"
         def exc = thrown(HttpClientErrorException)
         exc.rawStatusCode == 400
         exc.responseBodyAsString.contains("Switch '$switchId' is in 'Active' state")
     }
 
-    def "Cannot delete inactive switch with active ISL"() {
+    @Unroll
+    def "Unable to delete an inactive switch with #linkType ISLs"() {
         requireProfiles("virtual")
 
-        given: "An inactive switch with active ISL"
-        def isl = northbound.activeLinks[0]
-        def switchId = isl.source.switchId
-        lockKeeper.knockoutSwitch(switchId)
-        Wrappers.wait(WAIT_OFFSET) {
-            !northbound.activeSwitches.any { it.switchId == switchId}
-        }
-
-        when: "Try to delete inactive switch with active ISL"
-        northbound.deleteSwitch(switchId, false)
-
-        then: "Got 400 BadRequest because switch has active ISL"
-        def exc = thrown(HttpClientErrorException)
-        exc.rawStatusCode == 400
-        exc.responseBodyAsString.matches(".*Switch '$switchId' has \\d+ active links\\. Unplug and remove them first.*")
-
-        and: "Cleanup: Activate the switch back"
-        lockKeeper.reviveSwitch(switchId)
-        Wrappers.wait(WAIT_OFFSET) {
-            northbound.activeSwitches.any { it.switchId == switchId }
-        }
-    }
-
-    def "Cannot delete inactive switch with inactive ISLs"() {
-        requireProfiles("virtual")
+        given: "An inactive switch with ISLs"
+        def sw = topology.getActiveSwitches()[0]
+        def swIsls = topology.getRelatedIsls(sw)
 
-        given: "An inactive switch with inactive ISLs"
-        def switchId = topology.getActiveSwitches()[0].dpId
-        // get all active ISLs on switch
-        def activeSwitchLinks = northbound
-                .getAllSwitchLinks(switchId)
-                .findAll { it.state == IslChangeType.DISCOVERED }
-        // deactivate all active ISLs on switch
-        downLinkPorts(activeSwitchLinks)
-        Wrappers.wait(WAIT_OFFSET) {
-            northbound.getAllSwitchLinks(switchId).every { it.state != IslChangeType.DISCOVERED }
+        if (linkType == "inactive") {
```

Comment:
too many `if`s in this test for no good reason. Instead of branching inside test it is better to create a separate test. I think we should avoid having `if`s in tests at all (which we actually do, if you search our specs for `if`s, you'll get close to 0 results), test case should be a valid set of steps with no branching, every branch is just another test case. Sometimes a use of small `if` is fine and justified, but I don't think that this case falls into such exception, we are trying to mix different steps to save some lines of code

Reply:
Fixed.